### ISSUE TITLE
Front-End Attachment Feature Missing and OpenAI API Key Connection Error in Application

### ISSUE THREAD

**Issue by 1195srijan** [microsoft/TaskWeaver#350]
On frontend I am not seeing Attach files feature 

The OpenAI API key is causing a connection error in task weaver, although it works correctly in another.

Screenshots attached:
![task-error2](https://github.com/microsoft/TaskWeaver/assets/114597830/448bff3e-5b02-4e6e-a2b8-001d04c6809d)
![task-error2](https://github.com/microsoft/TaskWeaver/assets/114597830/448bff3e-5b02-4e6e-a2b8-001d04c6809d)
![task-error2](https://github.com/microsoft/TaskWeaver/assets/114597830/448bff3e-5b02-4e6e-a2b8-001d04c6809d)
![task-error](https://github.com/microsoft/TaskWeaver/assets/114597830/3700e85e-d9f9-42dc-b6cf-696c46d2e860)
![task-error](https://github.com/microsoft/TaskWeaver/assets/114597830/3700e85e-d9f9-42dc-b6cf-696c46d2e860)
![task-error](https://github.com/microsoft/TaskWeaver/assets/114597830/3700e85e-d9f9-42dc-b6cf-696c46d2e860)

**Comment by liqul** [comment]
Could you share more on your configurations?

**Comment by 1195srijan** [comment]
{
  "llm.api_base": "https://api.openai.com/v1",
  "llm.api_key": "sk-proj-xxxxxxxxxxx",
  "llm.model": "gpt-4-1106-preview"
}

If you need any specific file, let me know. This is from taskweaver_config.json

**Comment by liqul** [comment]
The error is indicating that the connection between your agent and openai endpoint has a connection error. I suggest you check the environment issues such as the openai sdk version, urllib3 version, etc. Make sure you can access the endpoint programmly in any code.

**Comment by 1195srijan** [comment]
Now I am getting below error:

httpx.ConnectError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1007)

Tried to update Zscalar certificates in cacert.pem file but that is not working.

**Comment by 1195srijan** [comment]
Okay, this error is resolved, it was because there was some restrictions on office laptop. But still attach files feature is not seen on frontend. May I know what to do for that?

**Comment by liqul** [comment]
I cannot reproduce this issue. Could you check if there are error logs printed during the lauch of the frontend?

**Comment by 1195srijan** [comment]
No, I am not seeing any error. I am using model gpt-4-1106-preview is that an issue that I am not seeing attach file functionality?

**Comment by liqul** [comment]
> No, I am not seeing any error. I am using model gpt-4-1106-preview is that an issue that I am not seeing attach file functionality?

I don't think the model will affect the frontend. One thing you can check is the version of Chainlit.

**Comment by 1195srijan** [comment]
> > No, I am not seeing any error. I am using model gpt-4-1106-preview is that an issue that I am not seeing attach file functionality?
> 
> I don't think the model will affect the frontend. One thing you can check is the version of Chainlit.

Chainlit, version is 1.1.101. Do we need to do some configuration then attach files feature is visible on frontend or by default it should always show?

**Comment by liqul** [comment]
Yes, there is a configuration in `playground/UI/.chainlit/config.toml`. Check if you can find this line
```
multi_modal.enabled = true
```

If you clone the repo, this configuration should be there.

**Comment by liqul** [comment]
I think the issue is with the new chainlit version. We haven't got bandwidth to adapt to the new version. My local test was based on version 1.0.500.

**Comment by liqul** [comment]
Have a PR for this issue https://github.com/microsoft/TaskWeaver/pull/362. 
If you need to fix it before the PR merge, you can make the following change 

![image](https://github.com/microsoft/TaskWeaver/assets/7489260/6ae6f2c9-c8a2-49db-9dab-0b42d8449955)

Also, ensure that your chainlit version is below 1.1.300 as it has an incompatible update.

**Comment by 1195srijan** [comment]
Thanks, this worked. Can we also link the database? If yes, please guide me on using Microsoft SQL server database.
